package.json

```diff
@@ -24,7 +32,8 @@
     "build:cjs": "swc src --out-dir dist/cjs --config-file .swcrc.build.json --config module.type=commonjs",
     "build:clean": "rimraf dist && yarn build",
     "build:docs": "typedoc",
-    "build:esm": "swc src --out-dir dist/esm --config-file .swcrc.build.json --config module.type=es6",
+    "build:esm": "swc src --out-dir dist/esm --config-file .swcrc.build.json --config module.type=es6 && yarn build:esm:package",
+    "build:esm:package": "echo >dist/esm/package.json \"{\\\"type\\\":\\\"module\\\"}\"",
```

Comment:
Apparently Node.js doesn't allow using `.js` for both the CJS and ESM version in the same project (??), but you can work around it by adding a `package.json` to the ESM folder.